cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricDeleteQuery.java

```diff
@@ -37,7 +38,7 @@ public MetricDeleteQuery(long startTs, long endTs, String metricName,
     this.startTs = startTs;
     this.endTs = endTs;
     this.metricName = metricName;
-    this.sliceByTagValues = ImmutableMap.copyOf(sliceByTagValues);
+    this.sliceByTagValues = Maps.newHashMap(sliceByTagValues);
```

Comment:
this also answers your question, this is not hooked up to namespace delete yet, right?